### PR TITLE
Exclude If Present

### PR DESCRIPTION
**Pull request by shivansh02**
### Description

Exclude directories where specified files are present.

### Related Issue

#1863 

### How Has This Been Tested?

Tested manually by running backups. Selected filenames get appended to borg create command as "--exclude-if-present filename". Modified `test_exclusion_preview_populated` to test preview tab.
### Screenshots (if appropriate):
Screenshot from 2024-06-12 18-47-02
Screenshot from 2024-06-12 18-47-08

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] I have read the [CONTRIBUTING](https://vorta.borgbase.com/contributing/) guide.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

*I provide my contribution under the terms of the [license](./../LICENSE.txt) of this repository and I affirm the [Developer Certificate of Origin][dco].*

[dco]: https://developercertificate.org/

### CODE REVIEW

**Comment by m3nu** [comment]
Testing this locally now. In the future, you can also rebase a new PR if there were commits in the meantime.

**Comment by m3nu** [comment]
Works as expected locally and passes the correct args to Borg.

I would still suggest to add some common example files in the dialog, but keep them unselected. Like the ones shown in Borg's docs.

**Comment by shivansh02** [comment]
> I would still suggest to add some common example files in the dialog, but keep them unselected. Like the ones shown in Borg's docs.

I added two files (.nobackup and .vortaignore). Did these not show up for you?
I added the files as default value for exclude_if_present field in models.py and wrote a migration for the same. Did I miss something?

File: src/vorta/store/models.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ class BackupProfileModel(BaseModel):
     ssh_key = pw.CharField(default=None, null=True)
     compression = pw.CharField(default='lz4')
     exclude_patterns = pw.TextField(null=True)
-    exclude_if_present = pw.TextField(null=True)
+    exclude_if_present = pw.TextField(null=True, default="[] .nobackup\n[] .vortaignore")
```
Comment (m3nu) [inline src/vorta/store/models.py:82]: `.nobackup` is fine, but I wouldn't introduce a new standard, like `.vortaignore`. Why not add `CACHEDIR.TAG`, which is an existing [standard](https://bford.info/cachedir/) to support the related Borg flag.

File: src/vorta/store/migrations.py [hunk 2]
```diff
@@ -250,6 +250,17 @@ def run_migrations(current_schema, db_connection):
...
         )
 
+        if current_schema.version < 23:
+            _apply_schema_update(
+                current_schema,
+                23,
+                migrator.add_column_default(
+                    BackupProfileModel._meta.table_name,
+                    'exclude_if_present',
+                    pw.TextField(default="[] .nobackup\n[] .vortaignore"),
```
Comment (m3nu) [inline src/vorta/store/migrations.py:260]: See related comment for model

File: src/vorta/assets/UI/excludedialog.ui [hunk 3]
```diff
@@ -137,6 +177,7 @@
        </item>
       </layout>
      </widget>
+
```
Comment (m3nu) [inline src/vorta/assets/UI/excludedialog.ui:180]: Why are we adding a new line here?

**Review by m3nu — CHANGES_REQUESTED** [review]

File: src/vorta/store/models.py (hunk 1 above)
Comment (shivansh02) [reply src/vorta/store/models.py]: Wasn't aware about CACHEDIR.TAG, added it now.

File: src/vorta/views/exclude_dialog.py [hunk 4]
```diff
@@ -310,6 +352,73 @@ def preset_item_changed(self, item):
...
+                item.setCheckable(True)
+                item.setCheckState(Qt.CheckState.Checked)
+            elif pattern.startswith('[]'):
+                item.setText(pattern[3:])
+                item.setCheckable(True)
+                item.setCheckState(Qt.CheckState.Unchecked)
+            else:
+                item.setText(pattern)
+                item.setCheckable(True)
+                item.setCheckState(Qt.CheckState.Unchecked)
```
Comment (m3nu) [inline src/vorta/views/exclude_dialog.py:419]: Doesn't this mean that all existing exclude settings the user had are now disabled?

Comment (shivansh02) [reply src/vorta/views/exclude_dialog.py:419]: @m3nu That's true, I didn't take the existing ones into account. Should I try to make them checked by default?

Comment (m3nu) [reply src/vorta/views/exclude_dialog.py:419]: Yes, I would check them by default.

**Comment by m3nu** [comment]
I think we're good now. Let me run it again locally tonight.

**Review by m3nu — APPROVED** [review]

**Pull request merged**